File: iteration2/src/controllers/AdvisorController.java
package iteration2.src.controllers;

import java.util.ArrayList;
import java.util.List;

import iteration2.src.enums.ApprovalStatus;
import iteration2.src.models.Advisor;
import iteration2.src.models.SelectedCourse;
import iteration2.src.models.Student;

public class AdvisorController {

    private Advisor currentAdvisor;
    private List<Student> studentList;

    public AdvisorController(Advisor currentAdvisor) {
        this.currentAdvisor = currentAdvisor;
        this.studentList = currentAdvisor.fetchAdvisedStudents();
    }

    public void approveCourse(Student student, SelectedCourse selectedCourse) {
        currentAdvisor.acceptCourse(student, selectedCourse);
    }

    public void denyCourse(Student student, SelectedCourse selectedCourse) {
        currentAdvisor.rejectCourse(student, selectedCourse);
        selectedCourse.getCourseSection().decrementStudentCount();
    }

    public List<Student> getStudentListOrderByStatus() {

        List<Student> studentListOrder = new ArrayList<Student>();

        for (Student student : studentList) {
            if (student.getApprovalStatus() == ApprovalStatus.PENDING) {
                studentListOrder.add(student);
            }
        }

        for (Student student : studentList) {
            if (student.getApprovalStatus() == ApprovalStatus.DONE) {
                studentListOrder.add(student);
            }
        }
        return studentListOrder;
    }

    public List<String> getNotification() {
        return currentAdvisor.getNotifications();
    }

    public void logOut() {
        currentAdvisor.logout();
    }

    public void clearNotifications() {
        currentAdvisor.clearNotifications();
    }

    // GETTER AND SETTER
    public Advisor getCurrentAdvisor() {
        return currentAdvisor;
    }

    public void setCurrentAdvisor(Advisor currentAdvisor) {
        this.currentAdvisor = currentAdvisor;
    }
}

File: iteration3/src/controllers/AdvisorController.py
from iteration3.src.enums.ApprovalStatus import ApprovalStatus

class AdvisorController:

    def __init__(self, currentAdvisor):
        self._currentAdvisor = currentAdvisor
        self._studentList = currentAdvisor.fetchAdvisedStudents()

    def approveCourse(self, student, selectedCourse):
        return self._currentAdvisor.acceptCourse(student, selectedCourse)

    def denyCourse(self, student, selectedCourse):
        return self._currentAdvisor.rejectCourse(student, selectedCourse)



    def getStudentListOrderByStatus(self):
        studentListOrder = []

        for student in self._studentList:
            if(student.getApprovalStatus() == ApprovalStatus.PENDING):
                studentListOrder.append(student)
        
        for student in self._studentList:
            if(student.getApprovalStatus() == ApprovalStatus.DONE):
                studentListOrder.append(student)
        
        for student in self._studentList:
            if(student.getApprovalStatus() == ApprovalStatus.FINALIZED_REGISTRATION):
                studentListOrder.append(student)

        return studentListOrder
    




    def finalizeRegistration(self, student):
        self._currentAdvisor.finalizeRegistration(student)

    


    def getNotifications(self):
        return self._currentAdvisor.getNotifications()

    def logOut(self):
        self._currentAdvisor.logout()

    def clearNotifications(self):
        self._currentAdvisor.clearNotifications()
    
    def getCurrentAdvisor(self):
        return self._currentAdvisor
    
    def setCurrentAdvisor(self, currentAdvisor):
        self._currentAdvisor = currentAdvisor